test/fixture_init_test.cc

```diff
@@ -0,0 +1,35 @@
+#include "benchmark/benchmark.h"
+
+static bool have_entered_main = false;
+static int num_constructed = 0;
+
+class StaticInitFixtureTest : public ::benchmark::Fixture {
+ public:
+  StaticInitFixtureTest() {
+    ++num_constructed;
+    assert(have_entered_main);
+  }
+};
+
+BENCHMARK_DEFINE_F(StaticInitFixtureTest, ConstructionHappensAtRuntime)
+(benchmark::State &st) {
+  for (auto _ : st) {
+  }
+}
+
+BENCHMARK_REGISTER_F(StaticInitFixtureTest, ConstructionHappensAtRuntime)
+    ->Arg(42);
+
+BENCHMARK_F(StaticInitFixtureTest, ConstructionHappensAtRuntime2)
+(benchmark::State &st) {
+  for (auto _ : st) {
+  }
+}
+
+int main(int argc, char **argv) {
```

Comment:
maybe you can use googletest for this instead of asserts.

Reply:
OK, I will do it.

Reply:
Of the existing 35 testcases, only one(statistics_test) uses gtest. I add two testcases which does essentially the same thing, fixture_init_test.cpp doesn't use gtest while fixture_init_gtest.cpp does.

Reply:
is this file necessary any more?

gtest is a new addition so we're just starting to move testing over to it. you're ahead of the curve :)